packages/compass-schema-validation/src/modules/validation.js

```diff
@@ -382,65 +383,77 @@ export const fetchValidation = (namespace) => {
   return (dispatch, getState) => {
     const state = getState();
     const dataService = state.dataService.dataService;
-    let validation = {
-      validationAction: INITIAL_STATE.validationAction,
-      validationLevel: INITIAL_STATE.validationLevel
-    };
 
+    if (!dataService) {
+      return;
+    }
 
-    if (dataService) {
-      dataService.listCollections(
-        namespace.database,
-        { name: namespace.collection },
-        (errorColl, data) => {
-          if (errorColl) {
-            validation.error = errorColl;
+    dataService.listCollections(
+      namespace.database,
+      { name: namespace.collection },
+      (err, data) => {
+        const validation = validationFromCollection(err, data);
+
+        if (err) {
+          dispatch(zeroStateChanged(false));
+          dispatch(validationFetched(validation));
+          dispatch(isLoadedChanged(true));
+          return;
+        }
 
-            dispatch(zeroStateChanged(false));
-            dispatch(validationFetched(validation));
+        if (!validation.validator) {
+          validation.validator = '{}';
+          dispatch(validationFetched(validation));
+          dispatch(isLoadedChanged(true));
+          return;
+        }
 
-            return;
-          }
+        sendMetrics(
+          dispatch,
+          dataService,
+          namespace,
+          validation,
+          'schema-validation-fetched'
+        );
 
-          const options = data[0].options;
-
-          if (options) {
-            validation = defaults(
-              {
-                validator: options.validator,
-                validationAction: options.validationAction,
-                validationLevel: options.validationLevel
-              },
-              validation
-            );
-          }
+        validation.validator = EJSON.stringify(validation.validator, null, 2);
```

Comment:
This thing where it only stringifies validation.validator after passing it to sendMetrics was already there. sendMetrics fishes things out of there. Almost missed it because these actions that return a function that takes dispatch and getState aren't tested..